arch/ia32/jmp.S

```diff
@@ -5,127 +5,145 @@
  *
  * setjmp, longjmp
  *
- * Copyright 2018-2019 Phoenix Systems
- * Author: Jan Sikorski, Andrzej Glowinski
+ * Copyright 2018-2019, 2023 Phoenix Systems
+ * Author: Jan Sikorski, Andrzej Glowinski, Hubert Badocha
  *
  * This file is part of Phoenix-RTOS.
  *
  * %LICENSE%
  */
 
+ /* jmp_buf layout:
+ * esi
+ * ebx
+ * edi
+ * ebp
+ * esp + 4
+ * return address
+ * signal mask indicator (0 - not saved, otherwise saved(GOT addr on PIC))
+ * signal mask (if saved)
+ */
+
 #define __ASSEMBLY__
 
 
 .text
 
-.globl _setjmp
-.type _setjmp, %function
-_setjmp:
+
+.macro __SETJMP
```

Comment:
Maybe a better  name would be `SAVE_REGS`

Reply:
Yeah, changed it a bit